sdk/storage/Azure.Storage.Blobs.Cryptography/src/ClientSideDecryptionPolicy.cs

```diff
@@ -0,0 +1,373 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Cryptography;
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Specialized.Models;
+using Azure.Storage.Common;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.Blobs.Specialized
+{
+    /// <summary>
+    /// An <see cref="HttpPipelinePolicy"/> for appropriately downloading a blob using client-side encryption.
+    /// Adjusts the download range requested, if any, to ensure enough info is present to decrypt, as well as
+    /// pipes the returned content stream through a decryption stream. Should only be in pipelines for instances
+    /// of <see cref="EncryptedBlobClient"/>
+    /// </summary>
+    internal class ClientSideDecryptionPolicy : HttpPipelinePolicy
```

Comment:
Nit - the casing of `ClientSide` doesn't match the other `Clientside`s, but I think we should match whatever pattern we follow for all of those names.